Steamfitter.Api/Services/TaskExecutionService.cs

```diff
@@ -537,6 +549,66 @@ private STT.Task<string> RunTask(TaskEntity taskToExecute, ResultEntity resultEn
             return task;
         }
 
+        private async STT.Task<string> HttpTaskTask(TaskEntity taskToExecute)
+        {
+            HttpResponseMessage response;
+            using (var scope = _scopeFactory.CreateScope())
+            {
+                var tokenResponse = await ApiClientsExtensions.GetToken(scope);
+                var actionParameters = JsonSerializer.Deserialize<HttpInputString>(taskToExecute.InputString);
+                var url = actionParameters.Url;
+                if (url.Contains("?"))
+                {
+                    url += $"&asuser={taskToExecute.UserId.ToString()}" ;
+                }
+                else
+                {
+                    url += $"?asuser={taskToExecute.UserId.ToString()}" ;
+                }
+                var client = ApiClientsExtensions.GetHttpClient(_httpClientFactory, url, tokenResponse);
+                switch (taskToExecute.Action)
+                {
+                    case TaskAction.http_get:
+                        {
+                            response = await client.GetAsync(url);
+                            break;
+                        }
+                    case TaskAction.http_post:
+                        {
+                            StringContent content = new StringContent(actionParameters.Body, Encoding.UTF8, "application/json");
+                            client.DefaultRequestHeaders.Add("Accept", "application/json");
+                            response = await client.PostAsync(taskToExecute.VmMask, content);
```

Comment:
Why use taskToExecute.VmMask here?

Reply:
Wow!  I don't know how that got in there originally.  It should be url or EmptyString, because the default url for the client is set with url above.  It was working, beause VmMask was always an empty string.